### DISCUSSION TITLE
1.5.1-beta1

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#495] [Обновления]
## Список изменений
- Добавлена поддержка Одноклассников
- Улучшена очистка описания на Youtube
- Улучшено получение ID видео для Dailymotion
- Другие мелкие улучшения кода

* Все изменения были внесены @SashaXser в #476

## What's Changed
* dailymotion fix, ok.ru support + some fixed by @SashaXser in https://github.com/ilyhalight/voice-over-translation/pull/476

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.5.0.5...1.5.1-beta1

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.5.1-beta1'>1.5.1-beta1</a>.</em>

**Comment by NullVerdict** [comment]
> ## Список изменений
> 
>     * Добавлена поддержка Одноклассников
> 
>     * Улучшена очистка описания на Youtube
> 
>     * Улучшено получение ID видео для Dailymotion
> 
>     * Другие мелкие улучшения кода
> 
> 
>     * Все изменения были внесены @SashaXser в [dailymotion fix, ok.ru support + some fixed #476](https://github.com/ilyhalight/voice-over-translation/pull/476)
> 
> 
> ## What's Changed
> 
>     * dailymotion fix, ok.ru support + some fixed by @SashaXser in [dailymotion fix, ok.ru support + some fixed #476](https://github.com/ilyhalight/voice-over-translation/pull/476)
> 
> 
> **Full Changelog**: [1.5.0.5...1.5.1-beta1](https://github.com/ilyhalight/voice-over-translation/compare/1.5.0.5...1.5.1-beta1)
> _This discussion was created from the release [1.5.1-beta1](https://github.com/ilyhalight/voice-over-translation/releases/tag/1.5.1-beta1)._

Яндекс будет против переводов сайтов с пиратским контентом? Была идея добавить поддержку, но не знаю насколько это легально.

**Reply by ilyhalight** [reply]
> > ## Список изменений
> > ```
> > * Добавлена поддержка Одноклассников
> > 
> > * Улучшена очистка описания на Youtube
> > 
> > * Улучшено получение ID видео для Dailymotion
> > 
> > * Другие мелкие улучшения кода
> > 
> > 
> > * Все изменения были внесены @SashaXser в [dailymotion fix, ok.ru support + some fixed #476](https://github.com/ilyhalight/voice-over-translation/pull/476)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ## What's Changed
> > ```
> > * dailymotion fix, ok.ru support + some fixed by @SashaXser in [dailymotion fix, ok.ru support + some fixed #476](https://github.com/ilyhalight/voice-over-translation/pull/476)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > **Full Changelog**: [1.5.0.5...1.5.1-beta1](https://github.com/ilyhalight/voice-over-translation/compare/1.5.0.5...1.5.1-beta1)
> > _This discussion was created from the release [1.5.1-beta1](https://github.com/ilyhalight/voice-over-translation/releases/tag/1.5.1-beta1)._
> 
> Яндекс будет против переводов сайтов с пиратским контентом? Была идея добавить поддержку, но не знаю насколько это легально.

Хороший вопрос... скорее всего, перевод таких сайтов проживет пока кто-нибудь жалобу не напишет

**Reply by NullVerdict** [reply]
Может быть внедрить проверку кода страны и назначать проксирование в зависимости от него? В то же время можно объединить обычную версию с версией Cloudflare, но при этом назначать серверы Cloudflare только в случае блокировки серверов Яндекс.

**Reply by ilyhalight** [reply]
> Может быть внедрить проверку кода страны и назначать проксирование в зависимости от него? В то же время можно объединить обычную версию с версией Cloudflare, но при этом назначать серверы Cloudflare только в случае блокировки серверов Яндекс.

Не всегда cloudflare версия используется при заблокированных серверах, она так же может быть использована для некоторых расширений-загрузчиков (violetmonkey, firemonkey и т.д.). В теории это конечно можно проверять, но не знаю стоит ли так усложнять

**Reply by NullVerdict** [reply]
M3U8Proxy на FOSWLY не будет работать, потому что там не исправлена ошибка, из-за которой воркер постоянно брал url из первого запроса, что ломало ссылку на плейлист.

**Comment by NullVerdict** [comment]
Нашел три бага, они все актуальны с 1.5.0.
Первый баг заключается в том, что при заходе на DailyMotion, база данных инициализируется три раза подряд. 
Второй баг заключается в том, что если из-за автоперевода видео перевод запустится два раза, то настройка громкости на YouTube будет обнулена каждый раз, когда вы останавливаете перевод или переключаете видео.
Третий баг заключается в том, что кнопка выключения перевода не работает на стримах.
